### PR TITLE
Test against Ruby 2.2 and JRuby 2.0/2.1 modes

### PR DESCRIPTION
**Pull request by O-I**
Hi.

This attempts to move issues #348 and #374 forward. I've added JRuby 2.0 and 2.1 modes to the Travis build and allow them to fail. I've also added MRI 2.2 to see if it passes.

Cheers,
O-I

### CODE REVIEW

**Comment by O-I** [comment]
Ruby 2.2 fails with this [error](https://travis-ci.org/lostisland/faraday/jobs/49046561). I've removed it from .travis.yml for now. Not too confident I can troubleshoot that alone. I can explore. Or someone with more experience can enlighten me at their convenience. :smiley:

**Comment by mislav** [comment]
It seems that old Typheus fails to compile on Ruby 2.2. You could put back 2.2 on the CI list but figure out how to skip the old Typhoeus test. It's not crucial.

**Comment by O-I** [comment]
OK. So I think I figured out how to skip the Typhoeus tests on Ruby 2.2 and get it passing on Travis, but now I've hit some sort of SSL(?) [error](https://travis-ci.org/lostisland/faraday/jobs/49234198) on Ruby 2.1.

File: test/helper.rb [hunk 1]
```diff
@@ -74,6 +74,10 @@ def self.rbx?
       defined? RUBY_ENGINE and 'rbx' == RUBY_ENGINE
     end
 
+    def self.ruby_22?
+      defined? RUBY_ENGINE and 'ruby' == RUBY_ENGINE and /^2\.2\.\d+/ =~ RUBY_VERSION
```
Comment (mislav) [inline test/helper.rb]: You should do `RUBY_VERSION > '2.2'` and you don't need any RUBY_ENGINE checks. We want this method to return true for every version from 2.2.0 forward. We don't want Typhoeus tests run on 2.3.

**Comment by mislav** [comment]
Intermittent failure. We sometimes get those. Restarted and the build passd.

**Comment by O-I** [comment]
Wow. Plain old string comparison really makes that method clear. Anything else?

**Comment by mislav** [comment]
Thanks!

**Pull request closed without merging**